code/game/atoms_movable.dm

```diff
@@ -131,6 +131,7 @@
 
 /atom/movable/proc/throw_at(atom/target, range, speed, mob/thrower, spin = TRUE, datum/callback/callback) //If this returns FALSE then callback will not be called.
 	. = TRUE
+	glide_size = world.icon_size / max((speed - GLIDE_SIZE_CONSTANT), world.tick_lag) * world.tick_lag
```

Comment:
Possibly in the non-movement handler cases, one should cache the old glide size, do the operation, and then reset it? Not sure about mobs exactly, but for objs there are many things which call step. And at least some mobs, like simple animals, use walk and such.

Reply:
There's no clean point to reset glide_size, which is the issue.

Reply:
Well, in this particular case, you can cache the value on the throw datum and reset when the throw is fully resolved in the SS. But in general, yes, that's a valid point.